### PR TITLE
Suggest correct camel case

### PR DESCRIPTION
**Pull request by mcardosos**
Fixes https://github.com/Azure/azure-openapi-validator/issues/106
PTAL @veronicagg

### CODE REVIEW

File: src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs [hunk 1]
```diff
@@ -375,9 +375,11 @@ public static string GetCamelCasedSuggestion(string name)
             => Regex.Replace(
                 name,
                 @"(?x)
-                    (?<=\p{Lu}) # after any uppercase character,
+                    ((?<=\p{Lu}) # after any uppercase character,
                     \p{Lu}+     # find a series of uppercase characters,
-                    (?=\p{Lu}|\z) # followed by an uppercase character or the end of the string
+                    (?=\p{Lu}|\z)) # followed by an uppercase character or the end of the string
+                    |(\A[A-Z]) # or it happens to be the first character
+                    |((?<=[a-z]){1}[A-Z]\z) # or it is an uppercase at the end, next to a lowercase
```
Comment (veronicagg) [inline src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: which case is this last one for? example?

File: src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs [hunk 2]
```diff
@@ -375,9 +375,11 @@ public static string GetCamelCasedSuggestion(string name)
             => Regex.Replace(
```
Comment (veronicagg) [inline src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: It looks like we need to update the comments for this method, to include the updated examples.

File: src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs (hunk 1 above)
Comment (mcardosos) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: `ManyThingS` becomes `manyThings`

File: src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs (hunk 2 above)
Comment (mcardosos) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: You mean the ones after each regex section?

Comment (veronicagg) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: lines https://github.com/Azure/azure-openapi-validator/pull/131/files/d018e7f04bb49b7add7b373f35eb15ad1f722fd9#diff-ca0f26932fbcc64f5f8b35b071880575R369 and https://github.com/Azure/azure-openapi-validator/pull/131/files/d018e7f04bb49b7add7b373f35eb15ad1f722fd9#diff-ca0f26932fbcc64f5f8b35b071880575R370

File: src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs (hunk 1 above)
Comment (veronicagg) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: I see, ok, honestly we could also keep it as it, but fine, it's a suggestion, so the user evaluates at the end anyway :)

File: src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs (hunk 2 above)
Comment (mcardosos) [reply src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs]: FIxed

**Pull request merged**